library/ssl_tls.c

```diff
@@ -7969,6 +7976,19 @@ int mbedtls_ssl_handshake_wrapup( mbedtls_ssl_context *ssl )
     }
 #endif /* MBEDTLS_SSL_SRV_C && !MBEDTLS_SSL_NO_SESSION_CACHE */
 
+    if( authmode == MBEDTLS_SSL_VERIFY_NONE )
+    {
+        if( authmode == MBEDTLS_SSL_VERIFY_NONE )
```

Comment:
I think you want to make `authmode` volatile and insert `mbedtls_platform_enforce_volatile_reads()` between the two checks, or the compiler will optimise the second away.

Reply:
Done